PlatformConfigurations/IEEE802.1xPortBasedAuthentication.md

```diff
@@ -0,0 +1,872 @@
+default_bits            = 2048
+input_password          = key_password
+output_password         = key_password
+...
+[server]
+countryName             = US
+stateOrProvinceName     = North Carolina
+localityName            = Raleigh
+organizationName        = Example Inc.
+emailAddress            = admin@example.org
+commonName              = "Example Server Certificate"
+...
+```
+
+4. Customize the client configuration in the `/etc/freeradius/3.0/certs/client.cnf`:
+
+```
+...
+[ CA_default ]
+default_days            = 365
+...
+[ req ]
+distinguished_name      = client
+default_bits            = 2048
+input_password          = password_on_private_key
+output_password         = password_on_private_key
+...
+[client]
+countryName             = US
+stateOrProvinceName     = North Carolina
+localityName            = Raleigh
+organizationName        = Example Inc.
+emailAddress            = user@example.org
+commonName              = user@example.org
+...
+```
+
+5. Once customized, create the certificates by running `make all`.
+
+On the FreeRADIUS-Server:
+
+```
+root@localhost:~# make all
+```
+
+Exchanging these certificates securely allows for many of the different authentication protocols available.
+
+### Configuring EAP
+
+Freeradius supports many different methods of Extensible Authentication Protocol. A few include:
+
+- **EAP-TLS** (transport layer security) uses a secure TLS connection to authenticate
+  clients by using certificates. To use EAP-TLS, you need TLS client certificates for each
+  network client and a server certificate for the server. Note that the same certificate
+  authority (CA) must have issued the certificates. Always use your own CA to create certificates
+  because all client certificates issued by the CA you use can authenticate to your FreeRADIUS server.
+
+- **EAP-TTLS** (tunneled transport layer security) uses a secure TLS connection as the outer
+  authentication protocol to set up the tunnel. The inner authentication then uses, for
+  example, the password authentication protocol (PAP) or challenge handshake authentication
+  protocol (CHAP). To use EAP-TTLS, you need a TLS server certificate.
+
+- **EAP-PEAP** (protected extensible authentication protocol) uses a secure TLS connection
+  as the outer authentication protocol to set up the tunnel. The authenticator authenticates
+  the certificate of the RADIUS server. Afterwards, the supplicant authenticates through the
+  encrypted tunnel by using Microsoft challenge handshake authentication protocol version 2
+  (MS-CHAPv2) or other methods.
+
+To configure the type of EAP to use, edit the `/etc/freeradius/3.0/mods-available/eap` file:
+
+Set the password of the private key in the `private_key_password` parameter:
+
+```
+eap {
+...
+  tls-config tls-common {
+...
+    private_key_password = key_password
+...
+  }
+}
+```
+
+For our configuration demo, we will use "whatever" from the default certificate configuraiton
+as the `key_password`:
```

Comment:
is it private_key_password?

Reply:
This is not clear. what does password has to do anything with "default certificate configuraiton"?

Reply:
I have reworded this section to clarify that the the `private_key_password` parameter is needed to decrypt TLS operations and should match the default `input_password` and `output_password` defined by the user in `/etc/freeradius/3.0/certs/server.cnf` when generating the certificates.

![image](https://github.com/user-attachments/assets/913e377b-fd56-47d0-9cea-bc86cb7f4574)